### PR TITLE
expression: implement vectorized evaluation for 'builtinIntervalRealSig'

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinIntervalRealSig, for #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinCompareFunc/builtinIntervalRealSig-VecBuiltinFunc-8         	  10000     15984 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinIntervalRealSig-NonVecBuiltinFunc-8      	   5000     37097 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinIntervalRealSig-VecBuiltinFunc#01-8      	  10000     19508 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinIntervalRealSig-NonVecBuiltinFunc#01-8   	   3000     39998 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinIntervalRealSig-VecBuiltinFunc#02-8      	   5000     25621 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinCompareFunc/builtinIntervalRealSig-NonVecBuiltinFunc#02-8   	   3000     46326 ns/op	       0 B/op	       0 allocs/op
```
### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_compare_vec_test.go [hunk 1]
```diff
@@ -44,6 +44,11 @@ var vecBuiltinCompareCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETInt, types.ETInt, types.ETInt}},
 		{retEvalType: types.ETReal, childrenTypes: []types.EvalType{types.ETReal, types.ETReal, types.ETReal}},
 	},
+	ast.Interval: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETReal, types.ETReal}},
```
Comment (Reminiscent) [inline expression/builtin_compare_vec_test.go]: In mysql5.7, the function ```INTERVAL(N,N1,N2,N3,...)``` needs to meet the following conditions that ```N < N1< N2 < N3```. See [here](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_interval) for more detail.

File: expression/builtin_compare_vec_test.go [hunk 2]
```diff
@@ -44,6 +44,11 @@ var vecBuiltinCompareCases = map[string][]vecExprBenchCase{
...
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETReal, types.ETReal, types.ETReal}},
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETReal, types.ETReal, types.ETReal, types.ETReal}},
```
Comment (Reminiscent) [inline expression/builtin_compare_vec_test.go]: In mysql5.7, the function ```INTERVAL(N,N1,N2,N3,...)``` needs to meet the following conditions that ```N < N1< N2 < N3```. See [here](https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_interval) for more detail.

Comment (tsthght) [reply expression/builtin_compare_vec_test.go]: ok, will add some additional testcases like N1 < N2 < N3 < ...

File: expression/builtin_compare_vec_test.go [hunk 3]
```diff
@@ -49,6 +49,11 @@ var vecBuiltinCompareCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETReal, childrenTypes: []types.EvalType{types.ETReal, types.ETReal, types.ETReal}},
 	},
 	ast.Interval: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETReal, types.ETReal}},
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETReal, types.ETReal, types.ETReal}},
```
Comment (Reminiscent) [inline expression/builtin_compare_vec_test.go]: I think we should delete the wrong case test. Line 55 is the same.

File: expression/builtin_compare_vec_test.go [hunk 4]
```diff
@@ -49,6 +49,11 @@ var vecBuiltinCompareCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETReal, childrenTypes: []types.EvalType{types.ETReal, types.ETReal, types.ETReal}},
 	},
 	ast.Interval: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETReal, types.ETReal}},
```
Comment (Reminiscent) [inline expression/builtin_compare_vec_test.go]: The number of ```Interval``` function parameters argument should greater than one. So we should delete this line.

Comment (tsthght) [reply expression/builtin_compare_vec_test.go]: this testcase does have two parameters~

File: expression/builtin_compare_vec_test.go (hunk 3 above)
Comment (tsthght) [reply expression/builtin_compare_vec_test.go]: ok, i will delete the wrong case test.

File: expression/builtin_compare_vec.go [hunk 5]
```diff
@@ -438,11 +438,36 @@ func (b *builtinIntervalIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Col
...
 
 func (b *builtinIntervalRealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalReal(b.ctx, input, buf); err != nil {
```
Comment (XuHuaiyu) [inline expression/builtin_compare_vec.go]: s/ `err :=` / `err = `

Comment (tsthght) [reply expression/builtin_compare_vec.go]: Got it, i have modified. Thank you~

**Pull request merged**